java/debugger.jpda/src/org/netbeans/modules/debugger/jpda/models/ShortenedStrings.java

```diff
@@ -130,7 +213,14 @@ static String getStringWithLengthControl(StringReference sr) throws InternalExce
         try {
             ReferenceType st = ObjectReferenceWrapper.referenceType(sr);
             ArrayReference sa = null;
+            //only applicable if the string implementation uses a byte[] instead
+            //of a char[]
+            boolean isUTF16 = false;
+            boolean isCompactImpl = false;
+            int hiByteShift = 0;
+            int lowByteShift = 0;
```

Comment:
Please move these down to the use-site (directly before the if in 318) and remove the initializer. The if will initialize them.